File: java-api/src/main/java/de/davelee/personalman/api/PaidUserRequest.java
package de.davelee.personalman.api;

import lombok.*;
import java.util.Map;

/**
 * This class is part of the PersonalMan REST API. It represents a request to the server to mark the employees
 * as paid with the amount supplied.
 * @author Dave Lee
 */
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class PaidUserRequest {

    // company associated with
    private String company;

    // a table of employee by username and the amount of pay they have received.
    private Map<String, Double> employeePayTable;

    // start date of date range that they were paid for in format dd-MM-yyyy
    private String startDate;

    // end date of date range that they were paid for in format dd-MM-yyyy
    private String endDate;

    // The token of the user to verify that they are logged in
    private String token;

}

File: server/src/users/requests/paiduser.request.ts
import { ApiProperty } from '@nestjs/swagger';

export class PaidUserRequest {

    // company associated with
    @ApiProperty()
    company: string;

    // a table of employee by username and the amount of pay they have received.
    @ApiProperty()
    employeePayTable: Map<string, number>;

    // start date of date range that they were paid for in format dd-MM-yyyy
    @ApiProperty()
    startDate: string;

    // end date of date range that they were paid for in format dd-MM-yyyy
    @ApiProperty()
    endDate: string;

    // The token of the user to verify that they are logged in
    @ApiProperty()
    token: string;

    constructor(company: string, employeePayTable: Map<string, number>, startDate: string, endDate: string, token: string) {
        this.company = company;
        this.employeePayTable = employeePayTable;
        this.startDate = startDate;
        this.endDate = endDate;
        this.token = token;
    }

}